src/main/java/org/owasp/wrongsecrets/challenges/ChallengesController.java

```diff
@@ -133,41 +170,53 @@ public String challenge(Model model, @PathVariable Integer id) {
       }
     }
     enrichWithHintsAndReasons(model);
-    includeScoringStatus(model, challenge.getChallenge());
-    addWarning(challenge.getChallenge(), model);
+    includeScoringStatus(model, challengeDefinition);
+    addWarning(challengeDefinition, model);
     fireEnding(model);
     return "challenge";
   }
 
-  @PostMapping(value = "/challenge/{id}", params = "action=reset")
+  @PostMapping(value = "/challenge/{name}", params = "action=reset")
```

Comment:
maybe skip "url" everywhere and "name" and just use "shortname" ?

Reply:
Should be `url` otherwise it becomes confusion wrt the yaml configuration

Reply:
Fixed